components/builder-api/src/server/resources/pkgs.rs

```diff
@@ -1142,7 +1158,7 @@ fn do_upload_package_finish(req: &HttpRequest,
                                        Body::from_message("ds:up:6"));
     }
 
-    let session = authorize_session(&req, None).unwrap(); // Unwrap Ok
+    let session = authorize_session(&req, None, RoleAction::Write, None).unwrap(); // Unwrap Ok
```

Comment:
Should this require maintainer role?

Reply:
Same as above maintainer question. Write functions require maintainer or higher.

Reply:
:thinking:  Personal preference here, but it feels like it's better to be explicit on requiring maintainer access.  Otherwise wouldn't be be relying on implementation details of `authorize_session` to guarantee that we've set levels correctly?   This may be related to my lack of understanding around `action` vs `min_role` though.

Reply:
In most cases I definitely agree, in this case, this isn't the first authorization call in this function call - its a holdover from the previous behavior where we would call this function later on in the function flow to get access to session data. Same thing goes here. We've preserved the role action because it gives us the quick validation without making the minimum role comparison. I don't know if it's actually any faster but its one fewer operation.